jenkins/image_building/dib_elements/ci-base/post-root.d/image_prepull

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+# IMAGE_NAME conflicts with IMAGE_NAME variable in metal3-dev-env, unset it
+unset IMAGE_NAME
+
+if [[ "${IMAGE_OS}" == "ubuntu" ]]; then
+    export CONTAINER_RUNTIME="docker"
+    # Install Docker
+    sudo mkdir -m 0755 -p /etc/apt/keyrings
+    echo 'y' | curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+    sudo echo \
+    "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+    $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list
+
+    sudo apt-get update
+    sudo apt-get install docker-ce docker-ce-cli containerd.io jq -y
+    sudo groupadd docker || true
+    sudo usermod -aG docker metal3ci || true
+    sudo systemctl enable docker
+    sudo systemctl restart docker 
+else 
+    # Install images for downloading container images for centos
+    sudo apt-get install podman -y
+fi
+
+M3_DENV_ORG="${M3_DENV_ORG:-Nordix}"
+M3_DENV_REPO="${M3_DENV_REPO:-metal3-dev-env}"
+M3_DENV_URL="${M3_DENV_URL:-https://github.com/${M3_DENV_ORG}/${M3_DENV_REPO}.git}"
+M3_DENV_BRANCH="${M3_DENV_BRANCH:-add-image-prepull-make-target-sunnat}"
+M3_DENV_ROOT="${M3_DENV_ROOT:-/tmp}"
+M3_DENV_PATH="${M3_DENV_PATH:-${M3_DENV_ROOT}/${M3_DENV_REPO}}"
+export FORCE_REPO_UPDATE="${FORCE_REPO_UPDATE:-true}"
+export CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+export IMAGE_OS="${IMAGE_OS:-Ubuntu}"
+export EPHEMERAL_CLUSTER="${EPHEMERAL_CLUSTER:-kind}"
+
+# TODO(Sunnatillo): When we copy container images metal3 dev image does not spin up
+# on openstack.
+
+# Install metal3 requirements
+# mkdir -p "${M3_DENV_ROOT}"
+# if [[ -d "${M3_DENV_PATH}" ]] && [[ "${FORCE_REPO_UPDATE}" == "true" ]]; then
+#   sudo rm -rf "${M3_DENV_PATH}"
+# fi
+# if [ ! -d "${M3_DENV_PATH}" ] ; then
+#   pushd "${M3_DENV_ROOT}"
+#   git clone "${M3_DENV_URL}"
+#   popd
+# fi
+# pushd "${M3_DENV_PATH}"
+# git checkout "${M3_DENV_BRANCH}"
+# git pull -r || true
+# make prepull_images
+# popd
+
+# if [[ "${CONTAINER_RUNTIME}" == "docker" ]]; then
+#   sudo mkdir -p "${TMP_BUILD_DIR}/mnt/var/lib/docker/"
+#   sudo cp -a /var/lib/docker/. "${TMP_BUILD_DIR}/mnt/var/lib/docker/"
+# else
+#   sudo mkdir -p "${TMP_BUILD_DIR}/mnt/var/lib/containers/storage/"
+#   sudo cp -a /var/lib/containers/storage/. "${TMP_BUILD_DIR}/mnt/var/lib/containers/storage/"
```

Comment:
I don't think this is a safe way to get the images and it could be why the image doesn't work correctly.
We should use `docker image save` instead.

Reply:
I looked that as a one option as well. But  we cannot run docker command in chroot.  `docker image load` part can be implemented on metal3-dev-env that only loads already saved images. But I did not want to put more logic to metal3-dev-env. Copying folders works and we will have images there. But there is other problem with the size of an image.

Reply:
Ok that makes sense. I'm still very afraid of this commented out code. It is copying much more than just images. If we are anyway not able to use this code, can we just remove it? If we figure out a way to make it work later than we can add it back.

Reply:
I removed it.